ui/components/app/user-preferenced-currency-display/user-preferenced-currency-display.component.js

```diff
@@ -89,7 +89,13 @@ export default function UserPreferencedCurrencyDisplay({
   );
 }
 
-const UserPreferencedCurrencyDisplayPropTypes = {
+const UserPreferencedCurrencyDisplay = React.memo(
```

Comment:
Memoizing this component. It doesn't solve the expensive hook calculations, but at least prevents some additional re-renders.